app/src/main/java/org/simple/clinic/newentry/form/DateOfBirthField.kt

```diff
@@ -14,13 +14,15 @@ class DateOfBirthField(
   override fun validate(value: String): List<ValidationError> {
     if (value.isBlank()) return listOf(MissingValue)
```

Comment:
I think this entire code block could be a lot simpler if it used a `MutableList` internally and converted to a `List` at the point of return. What do you think?

Reply:
We aren't accumulating errors, this is what I came up with.

```kotlin
    override fun validate(value: String): List<ValidationError> {
    val validationErrors = mutableListOf<ValidationError>()
    if (value.isBlank()) validationErrors.add(MissingValue)

    try {
      val parsedDate = LocalDate.parse(value, dateTimeFormatter)
      if (parsedDate > today) {
        validationErrors.add(DateIsInFuture)
      }
    } catch (e: DateTimeParseException) {
      validationErrors.add(InvalidDateFormat)
    }

    return validationErrors.toList()
  }
```

I guess we could wait until we are returning more than one error. WDYT?

Reply:
Yeah, not something we need to do right now.